src/Microsoft.DotNet.Tools.Builder/README.md

```diff
@@ -0,0 +1,32 @@
+dotnet-build
+===========
+
+**NAME** 
+dotnet-build -- Orchestrates the compilation of a project and all its dependencies.
+
+**SYNOPSIS**
+dotnet build[options]
+
+**DESCRIPTION**
+
+The build verb orchestrates the compilation of a project: it gathers the dependencies of a project and decides which to compile. 
+
+Users should invoke the Build verb when they want the entire dependency graph compiled, and Compile when they want only a specific project compiled.
+
+Before any compilation begins, the build verb analyzes the project and its dependencies for incremental safety checks. If all checks clear out, then build proceeds with incremental compilation of the project and its dependencies. Otherwise it falls back to non-incremental compilation. Via a profile flag users can choose to receive additional information on how they can improve their build times.
+
+All the projects in the dependency graph that need compilation must pass the following safety checks in order for the compilation process to be incremental:
+- not use pre / post compile scripts
+- not load compilation tools from PATH (e.g., resgen, compilers)
+- use only known compilers (csc, vbc, fsc)
+
```

Comment:
We want to be able to use dotnet compile native. This internally uses CL.exe and Link.exe on Windows. Is this a known compiler from your POV? On Unix, we use Clang to compile the MSIL to CPP files.